block/src/qcow/mod.rs

```diff
@@ -2027,6 +2033,50 @@ mod tests {
         assert_eq!(read_header.backing_file_path, header.backing_file_path);
     }
 
+    /// Create a qcow2 file with itself as its backing file.
+    ///
+    /// Without configuration `max_nesting_depth`, this will cause infinite recursion when loading
+    /// the file.
+    fn new_self_referential_qcow(path: &Path) -> Result<()> {
+        let header = QcowHeader::create_for_size_and_path(3, 0x10_0000, path.to_str())?;
+        let mut disk_file = RawFile::new(
+            File::create(path).expect("Failed to create image file."),
+            false,
+        );
+        header.write_to(&mut disk_file)?;
+        Ok(())
+    }
+
+    #[test]
+    fn max_nesting_backing() {
+        let test_dir = TempDir::new_with_prefix("/tmp/ch").unwrap();
+        let img_path = test_dir.as_path().join("test.img");
+
+        new_self_referential_qcow(img_path.as_path()).unwrap();
+
+        let err = create_disk_file(
+            File::open(img_path.as_path()).expect("Failed to open qcow image file"),
+            false,
+            MAX_NESTING_DEPTH,
```

Comment:
Intuitively, `MAX_NESTING_DEPTH` should be allowed, while `MAX_NESTING_DEPTH + 1` should fail. Let's adjust the test here once you address the comments above.